### ISSUE TITLE
Feature request: map_state Name

### ISSUE THREAD

**Issue by friendodevil** [nervetattoo/banner-card#33]
Please add the ability to change Name for the different states when map_state. E.g.:

      entities:
        - entity: switch.sonoff_irrigation21
          map_state: 
           "off": mdi:power
            name: "Sprinklers off"   <-- Being requested
            color: grey              <-- Being requested issue #24 
           "on": mdi:power-cycle
            name: "Sprinklers on"   <-- Being requested
            color: green            <-- Being requested issue #24 
          action:
            service: switch.toggle

**Comment by nervetattoo** [comment]
Do you mean the entity name that shows as a heading above the icon?

**Comment by friendodevil** [comment]
Yes, would be great to be able to change icon heading and color when state changes

**Comment by nervetattoo** [comment]
While it does make the potential config format more complex I can quite easily add support for overriding everything based on `map_state`. Right now each mapped state takes a single string which is the icon name, but it could easily support an object (key:value list) that would allow you to go crazy:

```yaml
 entities:
    - entity: switch.sonoff_irrigation21
      map_state: 
        "off":
          value: mdi:power
          name: "Sprinklers off"
          size: 2 <- We could even resize it based on state
          action: <- Or a state specific action?
```

This is in line with how I support deep configuration other places, namely having a string-based shorthand and a full object based format as well.

Heck, I'll get this in! Color support doesn't exist overall so that is something I need to consider separately, but if I do add it it would automatically be supported in `map_state` as well.

**Comment by WhistleMaster** [comment]
Thanks for the update !

I've tried to use the new object mapping for map_state, but I get the following:

<img width="996" alt="Screenshot 2019-08-18 at 10 09 59" src="https://user-images.githubusercontent.com/1544568/63221914-62eae480-c1a0-11e9-941e-6264277d1e86.png">

**Comment by friendodevil** [comment]
Same here

**Comment by WhistleMaster** [comment]
Fixed with v0.8.0, thanks !

**Comment by friendodevil** [comment]
Works :-) Thank you so much
Would love it if color is added
